agent/exec/container/executor.go

```diff
@@ -41,12 +42,41 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 		}
 	}
 
+	// add v1 plugins to 'plugins'
 	addPlugins("Volume", info.Plugins.Volume)
 	// Add builtin driver "overlay" (the only builtin multi-host driver) to
 	// the plugin list by default.
 	addPlugins("Network", append([]string{"overlay"}, info.Plugins.Network...))
 	addPlugins("Authorization", info.Plugins.Authorization)
 
+	// retrieve v2 plugins
+	v2plugins, err := e.client.PluginList(ctx)
+	if err != nil {
+		log.L.WithError(err).Warning("PluginList operation failed")
+	} else {
+		// add v2 plugins to 'plugins'
+		for _, plgn := range v2plugins {
+			for _, typ := range plgn.Config.Interface.Types {
+				if typ.Prefix == "docker" && plgn.Enabled {
+					plgnTyp := typ.Capability
```

Comment:
Is it a good idea to be coercing capability to type?

What if there are multi-capability plugins?